service/storage_service.cc

```diff
@@ -1906,48 +1908,171 @@ future<> storage_service::do_stop_ms() {
     });
 }
 
-future<> storage_service::node_ops_cmd_heartbeat_updater(node_ops_cmd cmd, node_ops_id uuid, std::list<gms::inet_address> nodes, lw_shared_ptr<bool> heartbeat_updater_done) {
-    std::string ops;
-    if (cmd == node_ops_cmd::decommission_heartbeat) {
-        ops = "decommission";
-    } else if (cmd == node_ops_cmd::removenode_heartbeat) {
-        ops = "removenode";
-    } else if (cmd == node_ops_cmd::replace_heartbeat) {
-        ops = "replace";
-    } else if (cmd == node_ops_cmd::bootstrap_heartbeat) {
-        ops = "bootstrap";
-    } else {
-        throw std::runtime_error(format("node_ops_cmd_heartbeat_updater: node_ops_cmd is not supported"));
+class node_ops_ctl {
+    std::unordered_set<gms::inet_address> nodes_unknown_verb;
+    std::unordered_set<gms::inet_address> nodes_down;
+    std::unordered_set<gms::inet_address> nodes_failed;
+
+public:
+    const storage_service& ss;
+    sstring desc;
+    locator::host_id host_id;
+    inet_address endpoint;
```

Comment:
Can we add comments explaining what are host_id and endpoint?

Reply:
In v3